test/unit/org/apache/cassandra/cql3/validation/operations/AlterTest.java

```diff
@@ -572,6 +576,102 @@ public void testDoubleWith() throws Throwable
         }
     }
 
+
+    @Test
+    public void testAlterTableWithMemtable() throws Throwable
+    {
+        createTable("CREATE TABLE %s (a text, b int, c int, primary key (a, b))");
+        assertSame(MemtableParams.DEFAULT.factory(), getCurrentColumnFamilyStore().metadata().params.memtable.factory());
+
+        assertRows(execute(format("SELECT memtable FROM %s.%s WHERE keyspace_name = ? and table_name = ?;",
+                                  SchemaConstants.SCHEMA_KEYSPACE_NAME,
+                                  SchemaKeyspaceTables.TABLES),
+                           KEYSPACE,
+                           currentTable()),
+                   row("default"));
+
+        alterTable("ALTER TABLE %s WITH memtable = 'skiplist' ;");
+        assertSame(SkipListMemtable.FACTORY, getCurrentColumnFamilyStore().metadata().params.memtable.factory());
+        assertTrue(getCurrentColumnFamilyStore().getTracker().getView().getCurrentMemtable() instanceof SkipListMemtable);
+
+        assertRows(execute(format("SELECT memtable FROM %s.%s WHERE keyspace_name = ? and table_name = ?;",
+                                  SchemaConstants.SCHEMA_KEYSPACE_NAME,
+                                  SchemaKeyspaceTables.TABLES),
+                           KEYSPACE,
+                           currentTable()),
+                   row("skiplist"));
```

Comment:
Would it make sense for the system table to always have the name of a concrete configuration? It's a little odd that, in this case, `default` and `skiplist` are exactly the same thing. Are we doing this explicitly so that we can have a new default picked up automatically on startup after simply changing the YAML at the node level, rather than across the whole cluster?

Reply:
According to the CEP-19 discussion, users want to try out new settings a subset of nodes at a time, and prefer to not have settings that are not overridable per node.

This design addresses these concerns and addresses all scenarios I could think of: we can gradually switch all tables to a different implementation by changing the node's default; we can assign a specific implementation to some tables and, to analyze or work around problems, switch only some nodes to a different one; we can create targeted per-node settings in heterogeneous deployments.